formats/image-dimension-stacking-plugin/src/polus/plugins/formats/image_dimension_stacking/dimension_stacking.py

```diff
@@ -0,0 +1,189 @@
+        for t in range(br.T):
+            for c in range(br.C):
+                for z in range(br.Z):
+                    for y in range(0, br.Y, chunk_size):
+                        y_max = min([br.Y, y + chunk_size])
+                        for x in range(0, br.X, chunk_size):
+                            x_max = min([br.X, x + chunk_size])
+                            if group_by == "c":
+                                tile = br[y:y_max, x:x_max, 0, c : c + 1, 0]
+
+                                bw[y:y_max, x:x_max, 0, di : di + 1, 0] = tile
+                            if group_by == "t":
+                                tile = br[y:y_max, x:x_max, 0, t : t + 1, 0]
+                                bw[y:y_max, x:x_max, 0, 0, di : di + 1] = tile
+
+                            if group_by == "z":
+                                tile = br[y:y_max, x:x_max, z : z + 1, 0, 0]
+                                bw[y:y_max, x:x_max, di : di + 1, 0, 0] = tile
+
+
+def dimension_stacking(
+    inp_dir: Path,
+    file_pattern: str,
+    group_by: str,
+    out_dir: Path,
+) -> None:
+    """Image dimension stacking.
+
+    This function enables to write stack image of dimensions (z, c, t).
+        inp_dir : Path to input directory containing images
+        file_pattern : Pattern to parse image files
+        group_by : A single string variable to group filenames by
+        out_dir : Path to output directory.
+
+    """
+    dimensions = []
+    input_files = []
+
+    fps = fp.FilePattern(inp_dir, file_pattern)
+    out_name = fps.output_name()
+
+    for fl in fps(group_by=group_by):
+        f1, f2 = fl
+        if f1[0][0] == group_by:
+            file_dim = f1[0][1]
+            file = f2[0][1][0]
+            input_files.append(file)
+            dimensions.append(file_dim)
+
+    # Get the number of layers to stack
+    dim_size = len(dimensions)
+
+    with BioReader(input_files[0]) as br:
+        metadata = br.metadata
+
+    with BioWriter(
+        out_dir.joinpath(out_name),
+        metadata=metadata,
+        max_workers=num_workers,
+    ) as bw:
+        # Adjust the dimensions before writing
+        if group_by == "c":
+            bw.C = dim_size
+        if group_by == "t":
+            bw.T = dim_size
+        if group_by == "z":
+            bw.Z = dim_size
+            bw.ps_z = z_distance(input_files[0])
+
+        starttime = time.time()
+
+        with preadator.ProcessManager(
+            name=f"Stacking images of {group_by} dimensions",
+            num_processes=num_workers,
+            threads_per_process=4,
+        ) as pm:
+            threads = []
+            for file, di in zip(input_files, range(0, dim_size)):
+                thread = pm.submit_thread(
+                    write_image_stack,
+                    file,
+                    di=di,
+                    group_by=group_by,
+                    bw=bw,
+                )
+                threads.append(thread)
+            pm.join_threads()
```

Comment:
This function will block and will not return until all jobs are finished. The following `tqdm` progress bar is useless.